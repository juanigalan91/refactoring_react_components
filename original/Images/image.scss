.images {
    display: flex;
    flex-direction: row;
    &__link {
        height: 200px;
        display: inline-block;
    }

    &__img {
        height: 100%;
    }

    .image {
        display: initial;
    }
}