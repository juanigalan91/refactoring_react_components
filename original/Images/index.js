import React from 'react';
import PropTypes from 'prop-types';

const namespace = 'images';
const Images = (props) => {
  const { images, title } = props;
  return (
    <div className={namespace}>
      <h1 key="title" className={`${namespace}__title`}>{title}</h1>
      <ul key={namespace} className={`${namespace}__list`}>
        {
          images.map((image) => {
            const baseClassName = 'image';

            return (
              <li className={baseClassName} key={image.id}>
                <a
                  href={image.link}
                  className={`${baseClassName}__link`}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                      src={image.src}
                      alt={image.title}
                      className={`${baseClassName}__img`}
                  />
                  <h3 className={`${baseClassName}__img_title`}>{image.title}</h3>
                </a>
              </li>
            );
          })
        }
      </ul>
    </div>
  );
};

Images.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape({
    link: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    src: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })).isRequired,
  title: PropTypes.string.isRequired
};

export default Images;
