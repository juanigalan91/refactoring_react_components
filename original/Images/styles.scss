.images {
    background-color: #fff;
    border-radius: 4px;
    box-shadow: 0 1px 1px 0 rgba(0,0,0,.1);
    padding: 20px;
    margin-bottom: 20px;
    overflow-x: scroll; 

    &__list {
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        text-align: center;
        margin: 0;
        padding: 0;
    }

    &__title {
        margin: 0 0 20px 0;
    }

    .image {
        display: initial;

        &__link {
            height: 200px;
            display: inline-block;
            text-decoration: none;
            color: black;
        }
    
        &__img {
            height: 100%;
        }
    
        &__img_title {
            display: block;
        }
    }
}