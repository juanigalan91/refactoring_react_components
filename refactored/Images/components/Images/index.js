import React from 'react';
import PropTypes from 'prop-types';
import Image from '../Image';

const namespace = 'images';
const Images = (props) => {
  const { images, title } = props;
  return (
        <div className={namespace}>
            <h1 key="title" className={`${namespace}__title`}>{title}</h1>
            <ul key={namespace} className={`${namespace}__list`}>
                {
                    images.map(image => <Image {...image} key={image.id} />)
                }
            </ul>
        </div>
  );
};

Images.propTypes = {
  images: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

export default Images;
