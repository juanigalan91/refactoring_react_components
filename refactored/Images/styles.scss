@import './components/Images/styles';
@import './components/Image/styles';