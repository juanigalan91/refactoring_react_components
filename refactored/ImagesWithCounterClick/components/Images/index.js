import React from 'react';
import PropTypes from 'prop-types';
import Image from '../Image';
import useTracking from '../../hooks/useTracking';

const namespace = 'images';
const Images = (props) => {
    const { images, title, notAvailableDisclaimer } = props;
    const [count, track] = useTracking('Times the image carousel was clicked:');

    return (
        <div className={namespace}>
            <h1 key="title" className={`${namespace}__title`}>
                {title}
            </h1>
            <span className={`${namespace}__visits`}>
                Visits:
                {count}
            </span>
            <ul key={namespace} className={`${namespace}__list`}>
                {
                    images.map(
                        image => <Image {...image} track={track} notAvailableDisclaimer={notAvailableDisclaimer} key={image.id} />
                    )
                }
            </ul>
        </div>
    );
};

Images.propTypes = {
    images: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    notAvailableDisclaimer: PropTypes.string.isRequired
};

export default Images;
