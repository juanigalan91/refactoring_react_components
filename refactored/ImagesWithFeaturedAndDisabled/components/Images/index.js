import React from 'react';
import PropTypes from 'prop-types';
import Image from '../Image';

const namespace = 'images';
const Images = (props) => {
    const { images, title, notAvailableDisclaimer } = props;

    return (
        <div className={namespace}>
            <h1 key="title" className={`${namespace}__title`}>
                {title}
            </h1>
            <ul key={namespace} className={`${namespace}__list`}>
                {
                    images.map(
                        image => <Image {...image} notAvailableDisclaimer={notAvailableDisclaimer} key={image.id} />
                    )
                }
            </ul>
        </div>
    );
};

Images.propTypes = {
    images: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    notAvailableDisclaimer: PropTypes.string.isRequired
};

export default Images;
