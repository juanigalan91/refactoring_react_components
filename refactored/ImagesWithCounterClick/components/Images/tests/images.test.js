import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render, cleanup, fireEvent } from 'react-testing-library';
import Images from '..';
import images from '../../../../../data';

afterEach(cleanup);

const component = (
    <Images
        images={images}
        title="Test title"
        notAvailableDisclaimer="not available"
    />
);

describe('Images component', () => {
  it('renders the images component with the list of images', () => {
    const renderer = new ShallowRenderer();
    expect(renderer.render(component)).toMatchSnapshot();
  });

  it('executes the tracking hook when the image is clicked and adds to visits', () => {
    const { getByAltText, container } = render(component);

    const textEl = getByAltText('Red Dead Redemption 2');
    fireEvent.click(textEl);

    const visits = container.querySelector('.images__visits');
    expect(visits.textContent).toBe('Visits:1');
  });

  it('does not execute hook since item is not available', () => {
    const { getByAltText, container } = render(component);

    const textEl = getByAltText('Assassins creed Odyssey');
    fireEvent.click(textEl);

    const visits = container.querySelector('.images__visits');
    expect(visits.textContent).toBe('Visits:0');
  });
});
