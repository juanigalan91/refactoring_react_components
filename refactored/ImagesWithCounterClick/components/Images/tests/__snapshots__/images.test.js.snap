// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images component renders the images component with the list of images 1`] = `
<div
  className="images"
>
  <h1
    className="images__title"
  >
    Test title
  </h1>
  <span
    className="images__visits"
  >
    Visits:
    0
  </span>
  <ul
    className="images__list"
  >
    <Image
      id={1}
      isAvailable={true}
      isFeatured={true}
      link="https://www.amazon.com/Red-Dead-Redemption-2-PlayStation-4/dp/B01M5DZ525/ref=sr_1_3?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-3"
      notAvailableDisclaimer="not available"
      src="https://m.media-amazon.com/images/I/81z80lFufgL._AC_UL872_FMwebp_QL65_.jpg"
      title="Red Dead Redemption 2"
      track={[Function]}
    />
    <Image
      id={2}
      isAvailable={false}
      isFeatured={false}
      link="https://www.amazon.com/Marvels-Spider-Man-PlayStation-4/dp/B01GW8YDLK/ref=sr_1_4?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-4"
      notAvailableDisclaimer="not available"
      src="https://m.media-amazon.com/images/I/81d6JU6g1pL._AC_UL872_FMwebp_QL65_.jpg"
      title="Marvels Spiderman"
      track={[Function]}
    />
    <Image
      id={3}
      isAvailable={true}
      isFeatured={true}
      link="https://www.amazon.com/Marvels-Spider-Man-PlayStation-4/dp/B01GW8YDLK/ref=sr_1_4?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-4"
      notAvailableDisclaimer="not available"
      src="https://m.media-amazon.com/images/I/51po2bu7VnL._AC_UL872_FMwebp_QL65_.jpg"
      title="God of war"
      track={[Function]}
    />
    <Image
      id={4}
      isAvailable={false}
      isFeatured={false}
      link="https://www.amazon.com/Assassins-Creed-Odyssey-PlayStation-4-Standard/dp/B07DKYN13M/ref=sr_1_5?keywords=ps4+games&qid=1551460416&s=gateway&sr=8-5"
      notAvailableDisclaimer="not available"
      src="https://m.media-amazon.com/images/I/81da48KsFpL._AC_UL872_FMwebp_QL65_.jpg"
      title="Assassins creed Odyssey"
      track={[Function]}
    />
  </ul>
</div>
`;
